app/controllers/btn/BTNAccountingPeriodController.scala

```diff
@@ -53,34 +57,64 @@ class BTNAccountingPeriodController @Inject() (
     (identify andThen getSubscriptionData andThen requireSubscriptionData andThen btnStatus.subscriptionRequest).async { implicit request =>
       val pillar2Id                 = request.subscriptionLocalData.plrReference
       val changeAccountingPeriodUrl = appConfig.changeAccountingPeriodUrl
-      val subAccountingPeriod       = request.subscriptionLocalData.subAccountingPeriod
       val accountStatus             = request.subscriptionLocalData.accountStatus.forall(_.inactive)
 
-      val accountingPeriods = {
-        val startDate = HtmlFormat.escape(dateHelper.formatDateGDS(subAccountingPeriod.startDate))
-        val endDate   = HtmlFormat.escape(dateHelper.formatDateGDS(subAccountingPeriod.endDate))
+      def accountingPeriods(startDate: LocalDate, endDate: LocalDate) = {
+        val start = HtmlFormat.escape(dateHelper.formatDateGDS(startDate))
+        val end   = HtmlFormat.escape(dateHelper.formatDateGDS(endDate))
 
         SummaryListViewModel(
           rows = Seq(
             SummaryListRowViewModel(
               key = "btn.returnSubmitted.startAccountDate",
-              value = ValueViewModel(HtmlContent(startDate))
+              value = ValueViewModel(HtmlContent(start))
             ),
             SummaryListRowViewModel(
               key = "btn.returnSubmitted.endAccountDate",
-              value = ValueViewModel(HtmlContent(endDate))
+              value = ValueViewModel(HtmlContent(end))
             )
           )
         )
       }
 
-      obligationsAndSubmissionsService
-        .handleData(pillar2Id, subAccountingPeriod.startDate, subAccountingPeriod.endDate)
+      val accountingPeriod: Future[AccountingPeriodDetails] = request.userAnswers.get(BTNChooseAccountingPeriodPage) match {
```

Comment:
```suggestion
      val accountingPeriodDetails: Future[AccountingPeriodDetails] = request.userAnswers.get(BTNChooseAccountingPeriodPage) match {
```
I recommend renaming this to `accountingPeriodDetails`.